setup.py

```diff
@@ -24,6 +24,7 @@ def read(file_name):
       keywords="mime magic file",
       license="MIT",
       test_suite='test',
+      python_requires='>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, !=3.5.*',
```

Comment:
With modern pip, that's right. Users on those will get the supported version for their running Python, ie. the currently released one.

Have you tested them lately with those versions? Things tend to slip if not tested regularly, and especially 3.0-3.2 did things in quite unique ways.

* 3.2 was removed ~4 years ago https://github.com/ahupp/python-magic/commit/62c3c17fb3c51c13f1a49c98e3a6526d6eef0adf
* 2.6 3.3-3.5 were removed ~1 year ago https://github.com/ahupp/python-magic/commit/e929976f67c116ba669554564303a4c8b7964dd1
* CI was added ~7 years ago and never tested 3.0-3-2 https://github.com/ahupp/python-magic/commit/029834f4d866f75dfa897bbfcafe2482a3675662

They're all EOL, except for 3.5. I'd suggest 3.5 could be kept, removed from `python_requires`, added to the CI (it should be possible to test 3.5 on Xenial, despite what https://github.com/ahupp/python-magic/commit/e929976f67c116ba669554564303a4c8b7964dd1 says), and add its Trove classifier.

Then only support non-EOL versions (but perhaps EOL 2.7 a while longer) and take full advantage of modern Python features and not worry about propping up removed and outdated things.

For example, here's the pip installs for python-magic from PyPI for December 2019, showing low numbers for EOL 2.6, 3.3-3.4:

| category | percent | downloads |
|----------|--------:|----------:|
| 2.7      |  64.25% |   988,918 |
| 3.6      |  16.57% |   255,128 |
| 3.7      |  13.80% |   212,434 |
| 3.5      |   2.74% |    42,197 |
| 3.8      |   1.21% |    18,620 |
| null     |   0.80% |    12,313 |
| 3.4      |   0.61% |     9,453 |
| 2.6      |   0.01% |       202 |
| 3.9      |   0.00% |         7 |
| 3.3      |   0.00% |         3 |
| Total    |         | 1,539,275 |

Source: `pip install -U pypistats && pypistats python_minor python-magic --last-month`

---

Alternatively just remove `python_requires` and it's pot luck for the few people using the old Pythons. They're probably already pinning to an old version anyway.

Reply:
Thanks, those are good points.  My main concern was breaking people who were successfully using this on 3.[1234], but it sounds like this will fall back to the last enabled version:

"If they do not match, it will attempt to install the last package distribution that supported that Python runtime."

https://packaging.python.org/guides/dropping-older-python-versions/

Will merge and then re-add support for 3.5.

Also, bummer how dominant 2.7 still is.

Reply:
Please see PR https://github.com/ahupp/python-magic/pull/202.